### PR TITLE
Convert several FCalls to QCalls, plus misc cleanup

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
This is converting the few methods both us have been staring at a lot in last few days, and also deletes the unused arguments.

File: src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs [hunk 1]
```diff
@@ -66,28 +65,6 @@ public static T[] GetSubArray<T>(T[] array, Range range)
...
-        public static object GetUninitializedObject(
-            // This API doesn't call any constructors, but the type needs to be seen as constructed.
-            // A type is seen as constructed if a constructor is kept.
-            // This obviously won't cover a type with no constructor. Reference types with no
-            // constructor are an academic problem. Valuetypes with no constructors are a problem,
-            // but IL Linker currently treats them as always implicitly boxed.
-            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]
-            Type type)
-        {
-            if (type is null)
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs:78]: Undid the sharing here. There was more lines duplicated than shared.

**Comment by am11** [comment]
Previously, `CurrentManagedThreadId` was changed back from QCall to FCall because it deoptimized the implementation, is it applicable to these QCalls? Would be nice if the underlying bottleneck is fixed in the JIT, so we don't ever need to go backwards (and potentially unblock other PRs like https://github.com/dotnet/runtime/pull/39474).

File: src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs [hunk 2]
```diff
@@ -138,15 +144,32 @@ public static int OffsetToStringData
...
+        {
+            if (type is null)
+            {
+                throw new ArgumentNullException(nameof(type), SR.ArgumentNull_Type);
+            }
+
+            if (!(type is RuntimeType rt))
+            {
+                throw new SerializationException(SR.Format(SR.Serialization_InvalidType, type));
+            }
```
Comment (stephentoub) [inline src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs:164]: Any benefit to moving the null check?
```suggestion
            if (type is not RuntimeType rt)
            {
                if (type is null)
                {
                    throw new ArgumentNullException(nameof(type), SR.ArgumentNull_Type);
                }
            
                throw new SerializationException(SR.Format(SR.Serialization_InvalidType, type));
            }
```

File: src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs [hunk 3]
```diff
@@ -91,10 +91,10 @@ public static partial class Activator
             if (type is null)
                 throw new ArgumentNullException(nameof(type));
 
-            if (type.UnderlyingSystemType is RuntimeType rt)
-                return rt.CreateInstanceDefaultCtor(publicOnly: !nonPublic, skipCheckThis: false, fillCache: true, wrapExceptions: wrapExceptions);
+            if (!(type.UnderlyingSystemType is RuntimeType rt))
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs:94]:
```suggestion
            if (type.UnderlyingSystemType is not RuntimeType rt)
```
?

File: src/mono/netcore/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.Mono.cs [hunk 4]
```diff
@@ -129,9 +131,26 @@ internal static bool ObjectHasReferences(object obj)
...
-            return GetUninitializedObjectInternal(new RuntimeTypeHandle((RuntimeType)type).Value);
+            if (type is null)
+            {
+                throw new ArgumentNullException(nameof(type), SR.ArgumentNull_Type);
+            }
+
+            if (!(type is RuntimeType rt))
+            {
+                throw new SerializationException(SR.Format(SR.Serialization_InvalidType, type));
+            }
```
Comment (stephentoub) [inline src/mono/netcore/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.Mono.cs:151]: Same question as earlier

**Review by stephentoub — APPROVED** [review]

**Pull request merged**